From: https://sourceware.org/cgi-bin/get-raw-msg?listname=binutils&date=2009-05&msgid=20090506212719.GA28875%40lucon.org
From binutils-return-59315-listarch-binutils=sources dot redhat dot com at sourceware dot org Wed May 06 21:27:30 2009
Return-Path: <binutils-return-59315-listarch-binutils=sources dot redhat dot com at sourceware dot org>
Delivered-To: listarch-binutils at sources dot redhat dot com
Received: (qmail 16536 invoked by alias); 6 May 2009 21:27:29 -0000
Received: (qmail 16515 invoked by uid 22791); 6 May 2009 21:27:26 -0000
X-SWARE-Spam-Status: No, hits=-1.0 required=5.0 	tests=AWL,BAYES_00,NO_DNS_FOR_FROM
Date: Wed, 6 May 2009 14:27:19 -0700
From: "H dot J dot  Lu" <hongjiu dot lu at intel dot com>
To: binutils at sources dot redhat dot com
Subject: PATCH: Support LD_AS_NEEDED
Message-ID: <20090506212719.GA28875@lucon.org>
Reply-To: "H dot J dot  Lu" <hjl dot tools at gmail dot com>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
Content-Disposition: inline
User-Agent: Mutt/1.5.18 (2008-05-17)
Mailing-List: contact binutils-help at sourceware dot org; run by ezmlm
Precedence: bulk
List-Id: <binutils.sourceware.org>
List-Subscribe: <mailto:binutils-subscribe at sourceware dot org>
List-Archive: <http://sourceware.org/ml/binutils/>
List-Post: <mailto:binutils at sourceware dot org>
List-Help: <mailto:binutils-help at sourceware dot org>, <http://sourceware dot org/ml/#faqs>
Sender: binutils-owner at sourceware dot org
Delivered-To: mailing list binutils at sourceware dot org

Hi,

I got a request to support LD_AS_NEEDED.  OK for trunk?

Thanks.


H.J.
----
2009-05-06  H.J. Lu  <hongjiu.lu@intel.com>

	* NEWS: Mention LD_AS_NEEDED.

	* ld.texinfo: Document LD_AS_NEEDED.

	* ldmain.c (main): Handle LD_AS_NEEDED.

----
Rebased, dropped NEWS part -- Victor Rodriguez <victor.rodriguez.bahena@intel.com>

diff --git a/ld/lexsup.c b/ld/lexsup.c
index 86a033a3..7a96f57d 100644
--- a/ld/lexsup.c
+++ b/ld/lexsup.c
@@ -612,6 +612,9 @@ parse_args (unsigned argc, char **argv)
 
   ldemul_add_options (is, &shortopts, il, &longopts, irl, &really_longopts);
 
+  if (getenv ("LD_AS_NEEDED") != NULL)
+    input_flags.add_DT_NEEDED_for_regular = TRUE;
+
   /* The -G option is ambiguous on different platforms.  Sometimes it
      specifies the largest data size to put into the small data
      section.  Sometimes it is equivalent to --shared.  Unfortunately,
